tests/cypress/integration/advantage_spec.js

```diff
@@ -17,4 +17,156 @@ context("Advantage", () => {
   });
+
+  it("should be able to start a free trial", () => {
+    const randomEmail = getRandomEmail();
+
+    cy.visit("/advantage/subscribe?test_backend=true");
+    cy.acceptCookiePolicy();
+    cy.findByRole("dialog").should("not.exist");
+    cy.scrollTo("bottom");
+    cy.findByLabelText(/Software and security updates only/).click({
+      force: true,
+    });
+    cy.findByText("Buy now").click();
+    cy.findByRole("dialog").should("be.visible");
+
+    // fill in the form
+    // fill in card information
+    cy.get("#card-element").within(() => {
+      return cy
+        .getWithinIframe(`[placeholder="Card number"]`)
+        .type("4242424242424242");
+    });
+
+    cy.get("#card-element").within(() => {
+      return cy.getWithinIframe(`[placeholder="MM / YY"]`).type("1230");
+    });
+
+    cy.get("#card-element").within(() => {
+      return cy.getWithinIframe(`[placeholder="CVC"]`).type("123");
+    });
+
+    cy.get("#card-element").within(() => {
+      return cy.getWithinIframe(`[placeholder="ZIP"]`).type("12345");
+    });
+
+    // fill in the rest of the form
+    cy.findByLabelText("Email my receipt to:").type(randomEmail);
+    cy.findByLabelText("Name:").type("Test Name");
+    cy.findByLabelText("Organisation:").type("Abbey Road Studios");
+    cy.findByLabelText("Address:").type("Abbey Road");
+    cy.findByLabelText("Postal code:").type("NW8 9AY");
+    cy.findByLabelText("Country/Region:").select("Austria");
+    cy.findByLabelText("City:").type("London");
+    cy.findByRole("button", { name: "Next step" }).click();
+
+    // wait for request to be sent
+    cy.intercept("POST", "/advantage/purchase-account*").as("purchaseAccount");
+    cy.intercept("POST", "/advantage/customer-info*").as("customerInfo");
+    cy.intercept("POST", "/advantage/subscribe/preview*").as("preview");
+
+    // assert that a matching request has been made
+    cy.wait("@purchaseAccount");
+    cy.wait("@customerInfo");
+    cy.wait("@preview");
+
+    // test for contents of 2nd step
+    cy.findByText("1 x $225.00");
+
+    cy.findByText("Use free trial month").click();
+    cy.findByLabelText(/I agree to the Ubuntu Advantage service terms/).click({
+      // Need to use { force: true } because the actual input element (radio button)
+      // that the label is for is invisible (we use our own styles)
+      // and cypress complains (it would usually indicate an issue, in our case it's intentional).
+      force: true,
+    });
+    cy.findByRole("button", { name: "Buy" }).click();
+
+    // The trial call is made
+    cy.intercept("POST", "/advantage/subscribe*").as("trial");
+    cy.wait("@trial").then((interception) => {
+      expect(interception.request.body.trialling).to.be.true;
+    });
+
+    // The user lands on the thank you page
+    cy.findByText("Thanks for your purchase");
+    cy.findByText(`We’ve sent your invoice to ${randomEmail}`);
+  });
+
+  it("should be able to buy a subscription", () => {
```

Comment:
This test is failing 

> AssertionError: Timed out retrying after 4000ms: Expected <div#react-root.p-modal__dialog> not to exist in the DOM, but it was continuously found.
      at Context.eval (http://0.0.0.0:8001/__cypress/tests?p=tests/cypress/integration/advantage_spec.js:121:29)
TestingLibraryElementError: Timed out retrying after 4000ms: Unable to find an element with the text: Thanks for your purchase.
https://github.com/canonical-web-and-design/ubuntu.com/pull/10515/checks?check_run_id=3753853576

Reply:
It's flaky, we might need to tweak the timeout to be longer than 4s

Reply:
I see. Instead of relying on timeouts it might be worth waiting for a specific event. Do you know what's happening when cypress times out? Might be worth enabling videos in cypress and taking a look.